addons/web/models/models.py

```diff
@@ -226,51 +228,507 @@ def cleanup(vals: dict) -> dict:
 
     @api.model
     @api.readonly
-    def web_read_group(self, domain, fields, groupby, limit=None, offset=0, orderby=False, lazy=True):
+    def web_read_group(self, domain, groupby=(), aggregates=(), having=(), offset=0, limit=None, order=None, empty_group=False):
         """
-        Returns the result of a read_group and the total number of groups matching the search domain.
+        Returns the result of a web_formatted_read_group and the total number of groups matching the search domain.
 
         :param domain: search domain
-        :param fields: list of fields to read (see ``fields``` param of ``read_group``)
-        :param groupby: list of fields to group on (see ``groupby``` param of ``read_group``)
-        :param limit: see ``limit`` param of ``read_group``
-        :param offset: see ``offset`` param of ``read_group``
-        :param orderby: see ``orderby`` param of ``read_group``
-        :param lazy: see ``lazy`` param of ``read_group``
+        :param groupby: list of groupby specification (see ``groupby``` param of ``web_formatted_read_group``)
+        :param aggregates: list of aggregate specification (see ``aggregates``` param of ``web_formatted_read_group``)
+        :param limit: see ``limit`` param of ``web_formatted_read_group``
+        :param offset: see ``offset`` param of ``web_formatted_read_group``
+        :param order: forced order, see ``order`` param of ``web_formatted_read_group``
+        :param bool empty_group: optional bool to active group_expand field feature with fold
+                group information.
         :return: {
             'groups': array of read groups
             'length': total number of groups
         }
         """
-        groups = self._web_read_group(domain, fields, groupby, limit, offset, orderby, lazy)
+        groups = self.web_formatted_read_group(domain, groupby, aggregates, having, offset, limit, order, empty_group)
 
         if not groups:
             length = 0
         elif limit and len(groups) == limit:
-            annotated_groupby = self._read_group_get_annotated_groupby(groupby, lazy=lazy)
             length = limit + len(self._read_group(
                 domain,
-                groupby=annotated_groupby.values(),
+                groupby=groupby,
                 offset=limit,
             ))
-
         else:
             length = len(groups) + offset
+
         return {
             'groups': groups,
-            'length': length
+            'length': length,
         }
 
     @api.model
-    def _web_read_group(self, domain, fields, groupby, limit=None, offset=0, orderby=False, lazy=True):
+    @api.readonly
+    def web_formatted_read_group(self, domain, groupby=(), aggregates=(), having=(), offset=0, limit=None, order=None, empty_group=False) -> list[dict]:
```

Comment:
I think that we should set some parameters as keyword-only. What do you think?
Same for other read_group functions including `_read_group`.
```suggestion
    def web_read_group(self, domain, groupby=(), aggregates=(), *, having=(), offset=0, limit=None, order=None, empty_group=False):
```

Also `empty_group` is undocumented.

Reply:
It sounds like a good idea for the Python usage (for _read_group maybe) but for not RPC method. Lot of language doesn't have keyword-only parameters then impose it from the outside world doesn't make sense for me.

Reply:
It is common to pass parameters by name in RPC calls. Even if a language does not have that ability, they have an API so that they can send a kind of dict to the call (or they generate positional bindings, but they send arguments by name).
That function is getting too many arguments to keep all by position.

Reply:
Fair enough but it should be another PR to make the change consistent on `_read_group`, `web_formatted_read_group`, `web_read_group`.

Reply:
Ok for me. Maybe it can be done before this PR?